### DISCUSSION TITLE
[Plugin redis] issues with CNAME SOA and ?

### DISCUSSION THREAD

**Discussion by Underknowledge** [coredns/coredns#6348] [Q&A]
Hi, now I'm bit sad. 

Had the fun idea to setup an DNS, you know the typical stuff you do on a Saturday evening. here I am after 3 days. 
But one where I dont have to write the same files to multiple servers, it should be cool! 
Seccondary and forward ar'nt too nice as when you loose your primary, nothing can survive a reboot. 

Oh, I can use redis? nice! 
well after figuring everything out I can proudly state, it is broken (could be me thou)
[Documentation](https://coredns.io/explugins/redis/): 

```
redis-cli> hgetall example.net.
 1) "_ssh._tcp.host1"
 2) "{\"srv\":[{\"ttl\":300, \"target\":\"tcp.example.com.\",\"port\":123,\"priority\":10,\"weight\":100}]}"
 3) "*"
 4) "{\"txt\":[{\"ttl\":300, \"text\":\"this is a wildcard\"}],\"mx\":[{\"ttl\":300, \"host\":\"host1.example.net.\",\"preference\": 10}]}"
 5) "host1"
 6) "{\"a\":[{\"ttl\":300, \"ip\":\"5.5.5.5\"}]}"
 7) "sub.*"
 8) "{\"txt\":[{\"ttl\":300, \"text\":\"this is not a wildcard\"}]}"
 9) "_ssh._tcp.host2"
10) "{\"srv\":[{\"ttl\":300, \"target\":\"tcp.example.com.\",\"port\":123,\"priority\":10,\"weight\":100}]}"
11) "subdel"
12) "{\"ns\":[{\"ttl\":300, \"host\":\"ns1.subdel.example.net.\"},{\"ttl\":300, \"host\":\"ns2.subdel.example.net.\"}]}"
13) "@"
14) "{\"soa\":{\"ttl\":300, \"minttl\":100, \"mbox\":\"hostmaster.example.net.\",\"ns\":\"ns1.example.net.\",\"refresh\":44,\"retry\":55,\"expire\":66},\"ns\":[{\"ttl\":300, \"host\":\"ns1.example.net.\"},{\"ttl\":300, \"host\":\"ns2.example.net.\"}]}"
```

what I have: 
```
 1) "xdl"
 2) "{\"cname\":[{\"host\":\"x.example.com\",\"ttl\":360}]}"
 3) "arpa.corp."
 4) "{\"soa\":[{\"expire\":67,\"minttl\":100,\"ns\":\"ns1WhoamI.arpa.corp.\",\"refresh\":44,\"retry\":55,\"ttl\":100}]}"
 5) "ns"
 6) "{\"cname\":[{\"host\":\"ns1.arpa.corp\",\"ttl\":360}]}"
 7) "ns3"
 8) "{\"a\":[{\"ip\":\"10.0.0.102\",\"ttl\":360}]}"
 9) "ns1"
10) "{\"a\":[{\"ip\":\"10.0.0.100\",\"ttl\":360}]}"
11) "ydl2"
12) "{\"a\":[{\"ip\":\"10.2.3.2\",\"ttl\":360}]}"
13) "ydl"
14) "{\"cname\":[{\"host\":\"x.example.com\",\"ttl\":360}]}"
15) "ns2"
16) "{\"a\":[{\"ip\":\"10.0.0.101\",\"ttl\":360}]}"
17) "@"
18) "{\"soa\":{\"expire\":67,\"minttl\":100,\"ns\":\"ns1WhoamI.arpa.corp.\",\"refresh\":44,\"retry\":55,\"ttl\":100}}"
19) "test"
20) "{\"a\":[{\"ip\":\"1.2.3.2\",\"ttl\":360}]}"
```

Querying SOA is fun
```
[root@CoreDNS02 ~ ] ❯ dig +short SOA @10.0.0.100 arpa.corp
;; Got bad packet: bad label type
106 bytes
8b 72 85 00 00 01 00 01 00 00 00 01 04 61 72 70          .r...........arp
61 04 63 6f 72 70 00 00 06 00 01 04 61 72 70 61          a.corp......arpa
04 63 6f 72 70 00 00 06 00 01 00 00 00 64 00 23          .corp........d.#
03 6e 73 31 04 61 72 70 61 04 63 6f 72 70 00 65          .ns1.arpa.corp.e
1b 19 79 00 00 00 2c 00 00 00 37 00 00 00 42 00          ..y...,...7...B.
00 00 64 00 00 29 04 d0 00 00 00 00 00 0c 00 0a          ..d..)..........
00 08 14 dc de c6 77 a3 c3 80                            ......w...
```

Cname is "fun" too. not usable with webservers and what not 
```
[root@CoreDNS03 ~ ] ❯ dig @10.0.0.100 ns.arpa.corp 
...
;; flags: qr aa rd; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 1
;; WARNING: recursion requested but not available
....
;; QUESTION SECTION:
;ns.arpa.corp.			IN	A

[root@CoreDNS03 ~ ] ❯ dig @10.0.0.100 ns.arpa.corp CNAME

; <<>> DiG 9.18.17 <<>> @10.0.0.100 ns.arpa.corp CNAME
...
;; flags: qr aa rd; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1
;; WARNING: recursion requested but not available
...
;; QUESTION SECTION:
;ns.arpa.corp.			IN	CNAME

;; ANSWER SECTION:
ns.arpa.corp.		300	IN	CNAME	ns1.arpa.corp.
```

When I use a config file, everything works as expected. 
So I think that's the plugins fault (or I'm a idiot, proved that many times already)

When you want to give it a try, be my guest! stuff is [Here](https://gitlab.com/Underknowledge/coredns-redis/-/tree/main) 
more or less, install redis, copy the config file .configfiles/redis/redis.conf into /etc/redis/redis.conf 
when you started it, add some records to redis. 
download the go script from [here](https://gitlab.com/Underknowledge/coredns-redis/-/jobs/5207274236)
and add some records to arpa.corp like [This](https://gitlab.com/Underknowledge/coredns-redis/-/raw/main/dnsmanager.go/README.md) 

download Coredns with the redis plug-in enabled from [here](https://gitlab.com/Underknowledge/coredns-redis/-/jobs/5207274234/artifacts/download?file_type=archive) and use it with 
[.configfiles/coredns/Corefile](https://gitlab.com/Underknowledge/coredns-redis/-/raw/main/.configfiles/coredns/Corefile)

TLDR: 
Redis behaves wired. CNAME'S dont get expanded, SOA gives bad packets back.
idk what's going on, but I might think there is an issue with the redis plugin. 

when you have ideas on how to get this to working, plzzz. would be sad when this have been a waste of time.

**Comment by chrisohaver** [comment]
https://github.com/codysnider/coredns-redis isn't maintained by the coredns org. You might try reaching out to the maintainers of that project.
